lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -514,14 +531,14 @@ export default function getRecipeInterface(
                     requestTransferMethod
                 );
             } catch (err) {
-                if (
-                    (err.type === STError.UNAUTHORISED && err.payload.clearTokens) ||
-                    err.type === STError.TOKEN_THEFT_DETECTED
-                ) {
-                    logDebugMessage(
-                        "refreshSession: Clearing tokens because of UNAUTHORISED or TOKEN_THEFT_DETECTED response"
-                    );
-                    clearSession(config, res, requestTransferMethod);
+                if (err.type !== STError.UNAUTHORISED || err.payload.clearTokens) {
```

Comment:
```suggestion
                if (err.type === STError.TOKEN_THEFT_DETECTED || err.payload.clearTokens) {
```

Reply:
This is done cause if there is a network error whilst querying the core, it should not clear the id refresh token if it exists